x-pack/plugins/maps/public/components/action_select.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Component } from 'react';
+import { EuiFormRow, EuiSuperSelect, EuiIcon } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { ActionExecutionContext, UiActionsActionDefinition } from 'src/plugins/ui_actions/public';
+
+interface Props {
+  value?: string;
+  onChange: (value: string) => void;
+  getFilterActions?: () => Promise<UiActionsActionDefinition[]>;
+  getActionContext?: () => ActionExecutionContext;
+}
+
+interface State {
+  actions: UiActionsActionDefinition[];
+}
+
+export class ActionSelect extends Component<Props, State> {
+  private _isMounted = false;
+  state: State = {
+    actions: [],
+  };
+
+  componentDidMount() {
+    this._isMounted = true;
+    this._loadActions();
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  async _loadActions() {
+    if (!this.props.getFilterActions || !this.props.getActionContext) {
+      return;
+    }
+    const actions = await this.props.getFilterActions();
+    if (this._isMounted) {
+      this.setState({ actions });
+    }
+  }
+
+  render() {
+    if (this.state.actions.length === 0 || !this.props.getActionContext) {
+      return null;
+    }
+
+    if (this.state.actions.length === 1 && this.props.value === this.state.actions[0].id) {
+      return null;
+    }
+
+    const actionContext = this.props.getActionContext();
+    const options = this.state.actions.map((action) => {
+      const iconType = action.getIconType ? action.getIconType(actionContext) : null;
```

Comment:
Are these methods ever optional?

From:
https://github.com/elastic/kibana/blob/7bd014abb387a4b6d896abf992c96f2ab98fad65/src/plugins/ui_actions/public/actions/action.ts#L74-L83

It seems like they shouldn't

Reply:
The typings are using `UiActionsActionDefinition`. Looks like it would be better to use `Action` instead. I can update and remove these checks.